gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/IntentController.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.controllers
+
+import com.netflix.spinnaker.gate.services.IntentService
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException
+import groovy.util.logging.Slf4j
+import io.swagger.annotations.ApiOperation
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@ConditionalOnProperty('services.keel.enabled')
+@RestController
+@RequestMapping(value = "/intents")
+@Slf4j
+class IntentController {
+
+  @Autowired
+  IntentService intentService
+
+  @ApiOperation(value = "Retrieve an intent")
+  @RequestMapping(value = "/{intentId}", method = RequestMethod.GET)
+  Map getIntent(@PathVariable(name = "intentId") String intentId) {
+    def result = intentService.getIntent(intentId)
+    if (!result){
+      log.warn("Intent {} not found", value("intent", intentId))
+      throw new NotFoundException("Intent not found (id: $intentId)")
+    }
+    return result
+  }
+
+  @ApiOperation(value = "Retrieve a list of intents")
+  @RequestMapping(method = RequestMethod.GET)
+  List<Map> getIntents(@RequestParam(value = "status", required = false) List<String> status) {
+    return intentService.getIntents(status)
+  }
+
+  @ApiOperation(value = "Delete an intent")
+  @RequestMapping(value = "/{intentId}", method = RequestMethod.DELETE)
+  Map deleteIntent(@PathVariable(name = "intentId") String intentId) {
+    return intentService.deleteIntent(intentId)
```

Comment:
I think modifications should go straight to keel, so not to have a circular dependency between keel & orca. Keel itself has pretty good auditing capabilities (who, when, what AND why), and also its own event stream that we can hook up into echo.

Reply:
Fair comment.

Do you think there's any desire on the part of a user to look at Task history and see intent modifications?

Otherwise I can get behind skipping `orca`, likely worth a small inline comment indicating the motivation since it's different than elsewhere.

Reply:
That's a good point - it would probably make sense to get modifications to show up in the Tasks history. What's our feeling on dummy marker task executions? I feel like we've done this before on a one-off for an internal team, but maybe an empty `TraceStage` or similar could be useful, then Keel would just fire-and-forget that orchestration into Orca?

Reply:
We had asked traffic to do it for `nimble` but they haven't actually gotten around to it.

If we would also like history to flow through the BDP, it might be trickier to try and avoid doing it in `orca`.

Pragmatically, if we want some of the things that going through `orca` would provide automatically, I don't _think_ the circular dependency is the end of world.

Reply:
I'll make the changes so that all these calls go through orca with some nice tasks.